### DISCUSSION TITLE
Cast datatype on filter

### DISCUSSION THREAD

**Discussion by emileindik** [PostgREST/postgrest#2790] [Q&A] [answered]
Is it possible to cast a column only when using it to filter?
```
http://localhost:3000/slot?select=startTimestamp,park(id,name)&startTimestamp::date=in.('2023-05-12','2023-05-12')`
```
Here "startTimestamp" is a timestamptz but I'd like to cast it as "date" so I can filter for those timestamps having certain dates.
However after viewing the db logs, the query does not seem to be trying to cast startTimestamp at all.

I considered creating a new view to handle all this but I'm afraid I'd lose my ability to do resource embedding with `park` in the view.

**Comment by steve-chavez** [comment, marked as answer]
Not possible because it would invalidate indexes. See this previous discussion https://github.com/PostgREST/postgrest/issues/1952
